### DISCUSSION TITLE
How to Hide tools section  for any role in pimcore 11

### DISCUSSION THREAD

**Discussion by sandyHM** [pimcore/pimcore#16675] [Q&A / Support] [answered]
How to Hide tools section  for any role in pimcore.

**Comment by sandyHM** [comment]
![tools](https://github.com/pimcore/pimcore/assets/107986459/c29435d0-38e6-41e7-bd59-07b21c2d8bdc)

**Comment by SamyMP** [comment]
Hi @sandyHM,

You have at least 3 ways to do that :
1. Create a perspective that hides this menu and set it as default for all users, and disable the default perspective for every roles. However, the admin will still have the possibility to access it

See the documentation here : https://pimcore.com/docs/platform/Pimcore/Tools_and_Features/Perspectives/

2. Add a custom css file in your project, load it in an `EventListener` : 

```php
// src/PimcoreAdminListener

namespace App\EventListener;

use Pimcore\Event\BundleManager\PathsEvent;

class PimcoreAdminListener
{
    public function addCSSFiles(PathsEvent $event)
    {
        $event->setPaths(
            array_merge(
                $event->getPaths(),
                [
                    '/static/css/your-custom-file.css',
                ]
            )
        );
    }
}
```

```yaml
# config/services.yaml

    # adds additional static files to admin backend
    App\EventListener\PimcoreAdminListener:
        tags:
            - { name: kernel.event_listener, event: pimcore.bundle_manager.paths.css, method: addCSSFiles }
```

And in your CSS file : 
```css
#pimcore_menu_extras {
    display: none;
}
```

3. Same as solution 2, but with Javascript : 
```js
document.querySelector('li#pimcore_menu_extras').style.display = 'none'
```

Please be aware of the security note in the documentation about perspectives :

> Security Note
Perspectives and Custom Views are not intended to be used to restrict access to data

This means you can "hide" the menu, but users could still access it if they have the rights to do so (like calling the route to clear the cache directly)

**Reply by sandyHM** [reply]
Hi @SamyMP thank you for you reply. will try .

**Reply by patriciomg** [reply, marked as answer]
Hi
well, I didn't need any extra css in order to hide those options. 
In Pimcore 11 you just need to install perspectives and then click on "hide.... "  option in a checkbox list :) 
regards,
P.

**Comment by sandyHM** [comment]
Thanks @patriciomg its working
